unit_race flaky test: vttablet/tabletserver/messager

Discovered in https://github.com/vitessio/vitess/pull/5651/checks?check_run_id=372705177

```
2020-01-03T17:09:32.0574180Z ok  	vitess.io/vitess/go/vt/vttablet/tabletmanager	1.154s
2020-01-03T17:09:33.2219506Z ok  	vitess.io/vitess/go/vt/vttablet/tabletserver/connpool	1.419s
2020-01-03T17:09:36.5306405Z --- FAIL: TestMessageManagerSend (0.01s)
2020-01-03T17:09:36.5308016Z ##[error]    message_manager_test.go:229: Message 1 is still present in inFlight cache
2020-01-03T17:09:36.5316783Z E0103 17:09:35.198986   18578 conn.go:724] Error reading packet from client 1 (@): read unix /tmp/fakesqldb114936372/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5365244Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5366783Z E0103 17:09:35.473935   18578 conn.go:724] Error reading packet from client 1 (@): read unix /tmp/fakesqldb289775107/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5367024Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5367459Z E0103 17:09:35.475400   18578 message_manager.go:520] Error sending messages: &{[] 0 0 [[VARBINARY("1") NULL]] <nil>}: non-eof
2020-01-03T17:09:36.5368131Z E0103 17:09:35.477158   18578 message_manager.go:513] Terminating connection due to error sending field info: non-eof
2020-01-03T17:09:36.5368608Z E0103 17:09:35.733404   18578 conn.go:724] Error reading packet from client 1 (@): read unix /tmp/fakesqldb779877863/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5368784Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5369230Z E0103 17:09:35.733461   18578 conn.go:724] Error reading packet from client 2 (@): read unix /tmp/fakesqldb779877863/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5369505Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5369959Z E0103 17:09:36.009317   18578 conn.go:724] Error reading packet from client 1 (@): read unix /tmp/fakesqldb326936602/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5370112Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5370551Z E0103 17:09:36.009456   18578 conn.go:724] Error reading packet from client 2 (@): read unix /tmp/fakesqldb326936602/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5370722Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5371165Z E0103 17:09:36.266728   18578 conn.go:724] Error reading packet from client 2 (@): read unix /tmp/fakesqldb330108593/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5371333Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5371764Z E0103 17:09:36.266766   18578 conn.go:724] Error reading packet from client 1 (@): read unix /tmp/fakesqldb330108593/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5371929Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5372364Z E0103 17:09:36.267096   18578 conn.go:724] Error reading packet from client 3 (@): read unix /tmp/fakesqldb330108593/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5372528Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5372953Z E0103 17:09:36.523883   18578 conn.go:724] Error reading packet from client 1 (@): read unix /tmp/fakesqldb213154396/fakesqldb.sock->@: use of closed network connection
2020-01-03T17:09:36.5373119Z io.ReadFull(header size) failed
2020-01-03T17:09:36.5373233Z FAIL
2020-01-03T17:09:36.5373422Z FAIL	vitess.io/vitess/go/vt/vttablet/tabletserver/messager	2.696s
2020-01-03T17:09:36.5373625Z ok  	vitess.io/vitess/go/vt/vttablet/tabletserver/planbuilder	1.116s
```

--- a/go/vt/vttablet/tabletserver/messager/message_manager_test.go
+++ b/go/vt/vttablet/tabletserver/messager/message_manager_test.go
@@ -219,6 +219,16 @@ func TestMessageManagerSend(t *testing.T) {
 		t.Errorf("Postpone: %s, want %v", got, want)
 	}
 
+	// Wait while the receiver is marked as busy.
+	for {
+		mm.mu.Lock()
+		busy := mm.receivers[0].busy
+		mm.mu.Unlock()
+		if !busy {
+			break
+		}
+	}
+
 	// Verify item has been removed from cache.
 	// Need to obtain lock to prevent data race.
 	mm.cache.mu.Lock()

